core/validatorapi/validatorapi.go

```diff
@@ -497,6 +499,103 @@ func (c Component) SubmitBlindedBeaconBlock(ctx context.Context, block *eth2api.
 	return nil
 }
 
+func (c Component) verifyValidatorRegistrationParSig(ctx context.Context, pubKey core.PubKey, slot eth2p0.Slot, randao eth2p0.BLSSignature) error {
+	// Calculate slot epoch
+	epoch, err := c.epochFromSlot(ctx, slot)
+	if err != nil {
+		return err
+	}
+
+	// Randao signing root is the epoch.
+	sigRoot, err := eth2util.EpochHashRoot(epoch)
+	if err != nil {
+		return err
+	}
+
+	return c.verifyParSig(ctx, core.DutyRandao, epoch, pubKey, sigRoot, randao)
+}
+
+func (c Component) verifyValidatorRegistrationSignature(ctx context.Context, registration *eth2api.VersionedSignedValidatorRegistration, pubkey core.PubKey, slot eth2p0.Slot) error {
+
+	epoch, err := c.epochFromSlot(ctx, slot)
+	if err != nil {
+		return err
+	}
+
+	var sig eth2p0.BLSSignature
+	switch registration.Version {
+	case spec.BuilderVersionV1:
+		if registration.V1.Signature == sig {
+			return errors.New("no V1 signature")
+		}
+		sig = registration.V1.Signature
+	default:
+		return errors.New("unknown version")
+	}
+
+	// Verify partial signature
+	// TODO: switch to registration.Root() when implemented on go-eth2-client (PR has been reaised)
+	sigRoot, err := registration.V1.Message.HashTreeRoot()
```

Comment:
Normally would used the Root() function 

This is implemented in go-eth2-client yet - PR has been raised 

Under the hood Root() just calles V1.Message.HashTreeRoot() so can use this in the interim until the PR is merged by Jim

https://github.com/attestantio/go-eth2-client/pull/23